generic-tabs/GenericTabs.js

```diff
@@ -35,11 +35,17 @@ export class GenericTabs extends SelectedMixin(BatchingElement) {
     return {
       selectors: {
         tabs: {
-          selector: el => el.querySelectorAll("[slot='tab']"),
+          selector: el =>
+            Array.from(el.children).filter(_ => _.matches('h1, h2, h3, h4, h5, h6, [slot="tab"]')),
           focusTarget: true,
         },
         panels: {
-          selector: el => el.querySelectorAll("[slot='panel']"),
+          selector: el =>
+            Array.from(el.children).filter(
+              _ =>
+                _.matches('h1 ~ *, h2 ~ *, h3 ~ *, h4 ~ *, h5 ~ *, h6 ~ *, [slot="panel"]') &&
+                !_.matches('h1, h2, h3, h4, h5, h6, [slot="tab"]'),
```

Comment:
There are different ways to do this, if it matters. If you prefer, here’s another way:

```suggestion
                 !(_ instanceof HTMLHeadingElement || _.slot === 'tab')
```